base/deprecated.jl

```diff
@@ -1806,6 +1806,11 @@ import .Iterators.enumerate
     return p
 end
 
+# ease transition for return type change of e.g. indmax due to PR #22907 when used in the
+# common pattern `ind2sub(size(a), indmax(a))`
+@deprecate(ind2sub(dims::NTuple{N,Integer}, idx::CartesianIndex{N}) where N,
+           ntuple(n -> idx[n], Val{N}()))
```

Comment:
Isn't this just `idx.I`?

Reply:
Yes, but I felt a bit uneasy advertising direct access to an internal field in the deprecation message.

Reply:
I like `Tuple(idx)` even if `CartesianIndex` objects aren't iterable.

Reply:
Changed to `Tuple(idx)`.